### PR TITLE
ISPN-10292 Add a schemas directory to the server from which "schemas"…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-10292

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/Extensions.java [hunk 1]
```diff
@@ -86,4 +93,16 @@ private void loadServerTasks(ClassLoader classLoader) {
...
    }
+
+   public void loadSchemas(String path, ProtobufMetadataManager protobufMetadataManager) throws Exception {
+      Path schemaPath = Paths.get(path);
+      if (schemaPath.toFile().exists()) {
+         List<Path> schemas = Files.walk(schemaPath)
+               .filter(f -> f.toString().endsWith(".proto"))
+               .collect(Collectors.toList());
+         for (Path schema : schemas) {
+            protobufMetadataManager.registerProtofile(schema.toString(), new String(Files.readAllBytes(schema)));
```
Comment (anistor) [inline server/runtime/src/main/java/org/infinispan/server/Extensions.java]: If there are any errors (after processing all the files) I would dump to log the names of the offending files and their errors and prevent the server from starting

**Comment by anistor** [comment]
The schemas directory approach does not work with server images so it is just half-useful. 
But there are many other issues with this approach which make it a bit dangerous

File: server/runtime/src/main/java/org/infinispan/server/Extensions.java (hunk 1 above)
Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/server/Extensions.java]: Unless the `schema` directory is mounted on a persistent volume, similar to how the `data` directory is

**Comment by tristantarrant** [comment]
Unless the `schema` directory is mounted on a persistent volume, similar to how the `data` directory is

**Comment by gustavonalle** [comment]
As commented in the JIRA, I think posting a .proto via REST or HotRod is better. Having it on disk is redundant, since the protobuf cache is already persistent and replicated, and this will create an extra knob adding unnecessary complexity

**Comment by tristantarrant** [comment]
Schemas can also be ingested into a server using a CLI batch file.

**Pull request closed without merging**